streamer/pipe.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A module that encapsulates all the platform-specific logic related to creating
+named pipes."""
+
+import os
+import uuid
+from threading import Thread
+from typing import Any
+
+class Pipe(Thread):
+  """A class that represents a pipe."""
+
+  def __init__(self):
+    """Initializes a non-functioning pipe."""
+    
+    self._read_pipe_name = ''
+    self._write_pipe_name = ''
+
+    # Windows specific declarations.
+    self.buf_size = 0
```

Comment:
It looks to me like this could also be private.  Or is it used outside the class somewhere I overlooked?